applications/callflow/src/module/cf_route_to_cid.erl

```diff
@@ -0,0 +1,91 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2011-2017, 2600Hz INC
+%%% @doc
+%%%
+%%% @end
+%%% @contributors
+%%%   Karl Anderson
+%%%   Kirill Sysoev
+%%%-------------------------------------------------------------------
+-module(cf_route_to_cid).
+
+-behaviour(gen_cf_action).
+
+-include("callflow.hrl").
+
+-export([handle/2]).
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% This module attempts to lookup device by it's internal cid number.
+%% Returns continue if fails to connect or stop when successful.
+%% @end
+%%--------------------------------------------------------------------
+-spec handle(kz_json:object(), kapps_call:call()) -> 'ok'.
+handle(Data, Call) ->
+    case find_device_by_cid_number(Call) of
+        {'ok', EndpointId} ->
```

Comment:
One immediate thought, without looking too deeply at the code, is the create a `Data` object with the EndpointId and call into `cf_device:handle/2`. So this module's only job is to resolve CallerID Number into an Endpoint ID.
Also, it could instead call into `cf_group:handle/2` and set "endpoints" to `["{ENDPOINT_ID}"]` which would handle users or devices.

Reply:
@jamesaimonetti Thanks for a hint. Decided to go cf_group:handle/2 way..